staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -43,10 +45,14 @@ type CustomResourceDefinitionVersion struct {
 	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
 	Name string
 	// Served is a flag enabling/disabling this version from being served via REST APIs
-	Served bool
-	// Storage flags the release as a storage version. There can be only one version
-	// flagged as Storage.
-	Storage bool
+	// Default is True.
+	// +optional
+	Served *bool
+	// Storage flags the version as a storage version. There must be exactly one flagged
+	// as storage version.
+	// Default is False.
+	// +optional
+	Storage *bool
```

Comment:
Same here, `+optional` is semantically wrong. Being optional and having a default are different things.

Reply:
if something is not optional, it provided by the user all the time and default does not make sense or I am missing something.